include/sb_handle/temp_memory_pool.h

```diff
@@ -0,0 +1,112 @@
+/***************************************************************************
+ *
+ *  @license
+ *  Copyright (C) Codeplay Software Limited
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  For your convenience, a copy of the License has been included in this
+ *  repository.
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *  portBLAS: BLAS implementation using SYCL
+ *
+ *  @filename temp_memory_pool.h
+ *
+ **************************************************************************/
+#ifndef TEMP_MEMORY_POOL_H
+#define TEMP_MEMORY_POOL_H
+
+#include <map>
+#include <mutex>
+
+namespace blas {
+class Temp_Mem_Pool {
+  using queue_t = cl::sycl::queue;
+  using event_t = std::vector<cl::sycl::event>;
+  using temp_usm_map_t = std::multimap<size_t, void*>;
+  using temp_usm_size_map_t = std::map<void*, size_t>;
+  using temp_buffer_map_t = std::multimap<size_t, cl::sycl::buffer<int8_t, 1>>;
+
+ public:
+  Temp_Mem_Pool(queue_t q)
+      : q_(q),
+        temp_buffer_map_tot_byte_size_(0),
+        temp_usm_map_tot_byte_size_(0) {}
```

Comment:
the `temp_usm_map_tot_byte_size_` is undefined if SB_ENABLE_USM is not defined.

Reply:
Fixed in https://github.com/codeplaysoftware/portBLAS/commit/aa1fdf1de65df04b8034d4949d3ba948a33fe2a6.